src/Features/VisualBasic/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.vb

```diff
@@ -1,177 +1,124 @@
-                {SyntaxKind.GreaterThanExpression, Tuple.Create(SyntaxKind.LessThanOrEqualExpression, SyntaxKind.LessThanEqualsToken)},
-                {SyntaxKind.GreaterThanOrEqualExpression, Tuple.Create(SyntaxKind.LessThanExpression, SyntaxKind.LessThanToken)},
-                {SyntaxKind.IsExpression, Tuple.Create(SyntaxKind.IsNotExpression, SyntaxKind.IsNotKeyword)},
-                {SyntaxKind.IsNotExpression, Tuple.Create(SyntaxKind.IsExpression, SyntaxKind.IsKeyword)}
-            }
-
-        Private Shared ReadOnly s_logicalInversesMap As Dictionary(Of SyntaxKind, Tuple(Of SyntaxKind, SyntaxKind)) =
-            New Dictionary(Of SyntaxKind, Tuple(Of SyntaxKind, SyntaxKind))(SyntaxFacts.EqualityComparer) From
-            {
-                {SyntaxKind.OrExpression, Tuple.Create(SyntaxKind.AndExpression, SyntaxKind.AndKeyword)},
-                {SyntaxKind.AndExpression, Tuple.Create(SyntaxKind.OrExpression, SyntaxKind.OrKeyword)},
-                {SyntaxKind.OrElseExpression, Tuple.Create(SyntaxKind.AndAlsoExpression, SyntaxKind.AndAlsoKeyword)},
-                {SyntaxKind.AndAlsoExpression, Tuple.Create(SyntaxKind.OrElseExpression, SyntaxKind.OrElseKeyword)}
-            }
-
-        Private Shared ReadOnly s_ifNodeAnnotation As New SyntaxAnnotation
-
-        Protected Shared Async Function FindIfNodeAsync(document As Document, cancellationToken As CancellationToken) As Task(Of SyntaxNode)
-            Dim root = Await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(False)
-            Dim result = root _
-                .GetAnnotatedNodesAndTokens(s_ifNodeAnnotation) _
-                .Single() _
-                .AsNode()
+        Protected Overrides Function GetRootWithInvertIfStatement(document As Document, model As SemanticModel, ifStatement As SyntaxNode, cancellationToken As CancellationToken) As SyntaxNode
 
-            Return result
-        End Function
+            Dim generator = SyntaxGenerator.GetGenerator(document)
+            Dim syntaxFacts = GetSyntaxFactsService()
 
-        Private Async Function InvertIfAsync(document As Document, node As SyntaxNode, cancellationToken As CancellationToken) As Task(Of Document)
-            Dim _updatedDocument = document
+            Dim result = UpdateSemanticModel(model, model.SyntaxTree.GetRoot().ReplaceNode(ifStatement, ifStatement.WithAdditionalAnnotations(s_ifNodeAnnotation)), cancellationToken)
 
-            ' Annotate the original node so we can get back to it.
-            Dim ifNode = node
-            _updatedDocument = Await _updatedDocument.ReplaceNodeAsync(ifNode, ifNode.WithAdditionalAnnotations(s_ifNodeAnnotation), cancellationToken).ConfigureAwait(False)
-            ifNode = Await FindIfNodeAsync(_updatedDocument, cancellationToken).ConfigureAwait(False)
+            Dim ifNode = FindIfNode(result.Root)
 
             ' Complexify the top-most statement parenting this if-statement if necessary
```

Comment:
can yuo explain why we need this for VB, but not for C#?

Reply:
This is needed for the following 2 tests in VB:
1)  `Dim x = Sub() If True Then Dim y Else Console.WriteLine(), z = 1` inverts to 
    `Dim x = (Sub() If False Then Console.WriteLine() Else Dim y), z = 1`
2) `Case Sub() If True Then Dim x Else Return, Nothing` inverts to
     `Case (Sub() If False Then Return Else Dim x), Nothing`

I'm trying to figure out if there is an equivalent in C# that also needs expanded.